diff --git a/src/pgrepr/src/value/apd.rs b/src/pgrepr/src/value/apd.rs
index 5a2e4b70c..b9603a1b9 100644
--- a/src/pgrepr/src/value/apd.rs
+++ b/src/pgrepr/src/value/apd.rs
@@ -224,6 +224,8 @@ fn test_to_from_sql_roundtrip() {
         let d_from_sql = Apd::from_sql(&Type::NUMERIC, &out).unwrap();
         assert_eq!(r.0, d_from_sql.0);
     }
+    inner("0");
+    inner("-0");
     inner("0.1");
     inner("0.0");
     inner("0.00");
@@ -231,12 +233,50 @@ fn test_to_from_sql_roundtrip() {
     inner("0.0000");
     inner("0.00000");
     inner("123456789.012346789");
+    inner("000000000000000000000000000000000000001");
+    inner("000000000000000000000000000000000000000");
+    inner("999999999999999999999999999999999999999");
     inner("123456789012345678901234567890123456789");
+    inner("-123456789012345678901234567890123456789");
     inner(".123456789012345678901234567890123456789");
+    inner(".000000000000000000000000000000000000001");
+    inner(".000000000000000000000000000000000000000");
+    inner(".999999999999999999999999999999999999999");
+    inner("-0.123456789012345678901234567890123456789");
     inner("1e25");
+    inner("-1e25");
     inner("9.876e-25");
     inner("-9.876e-25");
     inner("98760000");
     inner(".00009876");
+    inner("-.00009876");
     inner("NaN");
 } 
